File: uniapp/apis/membershipPackage.js
import request from '@/utils/request.js'

/**
 * 查询所有上架的套餐列表
 * 功能描述：查询所有上架状态的会员套餐
 * 入参：无
 * 返回参数：array - 套餐列表
 * url地址：/membership-package/list-online
 * 请求方式：GET
 */
export function getMembershipPackageListOnline(callbacks) {
  return request.get('/membership-package/list-online', null, callbacks)
}

/**
 * 根据会员类型查询套餐列表
 * 功能描述：查询指定会员类型的上架套餐列表
 * 入参：memberType - 会员类型（1-黄金，2-铂金）
 * 返回参数：array - 套餐列表
 * url地址：/membership-package/list-by-type
 * 请求方式：GET
 */
export function getMembershipPackageListByType(memberType, callbacks) {
  const params = memberType ? { memberType } : null
  return request.get('/membership-package/list-by-type', params, callbacks)
}

/**
 * 根据ID查询会员套餐
 * 功能描述：查询指定ID的会员套餐详情
 * 入参：id - 套餐ID
 * 返回参数：{ id, name, memberType, memberTypeDisplayName, durationDays, price, description, benefits, status, statusDisplayName, createTime, updateTime }
 * url地址：/membership-package/{id}
 * 请求方式：GET
 */
export function getMembershipPackageById(id, callbacks) {
  return request.get(`/membership-package/${id}`, null, callbacks)
}

File: springboot/src/main/java/org/example/springboot/entity/MembershipPackage.java
package org.example.springboot.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 会员套餐实体类
 * @author system
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("gym_membership_package")
@Schema(description = "会员套餐实体类")
public class MembershipPackage {

    @TableId(type = IdType.AUTO)
    @Schema(description = "套餐ID")
    private Long id;

    @Schema(description = "套餐名称")
    private String name;

    @Schema(description = "会员类型：1-黄金会员，2-铂金会员")
    @TableField("member_type")
    private Integer memberType;

    @Schema(description = "有效天数")
    @TableField("duration_days")
    private Integer durationDays;

    @Schema(description = "价格")
    private BigDecimal price;

    @Schema(description = "套餐描述")
    private String description;

    @Schema(description = "会员权益描述")
    private String benefits;

    @Schema(description = "状态：0-下架，1-上架")
    private Integer status;

    @Schema(description = "创建时间")
    @TableField("create_time")
    private LocalDateTime createTime;

    @Schema(description = "更新时间")
    @TableField("update_time")
    private LocalDateTime updateTime;

    /**
     * 是否上架
     */
    public boolean isOnline() {
        return status != null && status == 1;
    }

    /**
     * 是否黄金会员套餐
     */
    public boolean isGoldPackage() {
        return memberType != null && memberType == 1;
    }

    /**
     * 是否铂金会员套餐
     */
    public boolean isPlatinumPackage() {
        return memberType != null && memberType == 2;
    }

    /**
     * 获取会员类型显示名称
     */
    public String getMemberTypeDisplayName() {
        if (memberType == null) {
            return "未知";
        }
        switch (memberType) {
            case 1:
                return "黄金会员";
            case 2:
                return "铂金会员";
            default:
                return "未知";
        }
    }

    /**
     * 获取状态显示名称
     */
    public String getStatusDisplayName() {
        if (status == null) {
            return "未知";
        }
        switch (status) {
            case 0:
                return "下架";
            case 1:
                return "上架";
            default:
                return "未知";
        }
    }
}
